server/aws/mc-lambda.tf

```diff
@@ -0,0 +1,113 @@
+##
+#  Mectrics Collection Lambda
+##
+
+resource "aws_lambda_function" "metrics" {
+  function_name = var.service_name
+  description   = var.lambda-description
+#  filename      = "/tmp/lambda_validate_deploy.rb.zip"
+  s3_bucket     = var.lambda_code
+  s3_key        = var.lambda-function-code
```

Comment:
We either need to create this bucket and load it as part of this PR or we should leverage the archive_file provider like we do with the lambda used for Blue/Green deploy as can be seen here: https://github.com/cds-snc/covid-alert-server-staging-terraform/blob/master/server/aws/codedeploy.tf